prov/efa/src/rxr/rxr_ep.c

```diff
@@ -506,12 +506,26 @@ struct rxr_tx_entry *rxr_ep_alloc_tx_entry(struct rxr_ep *rxr_ep,
 }
 
 void rxr_inline_mr_reg(struct rxr_domain *rxr_domain,
-		       struct rxr_tx_entry *tx_entry)
+		       struct rxr_tx_entry *tx_entry, bool is_shm)
 {
 	ssize_t ret;
 	size_t offset;
-	int index;
+	int index = 0;
 
+	if (is_shm) {
+		tx_entry->iov_mr_start = index;
+		while (index < tx_entry->iov_count) {
+			ret = fi_mr_reg(&rxr_domain->util_domain.domain_fid,
+					tx_entry->iov[index].iov_base,
+					tx_entry->iov[index].iov_len,
+					FI_REMOTE_READ, 0, 0, 0,
```

Comment:
Is this the only access you'd need for this registration?

Reply:
Yes. It's only used for the peer's RMA Read operation.

Reply:
Can you reuse the registration in this function?

Reply:
Do you mean reuse the following function and pass different arguments for struct *fid_domain and int access?
```
545                         ret = fi_mr_reg(rxr_domain->rdm_domain,
546                                         tx_entry->iov[index].iov_base,
547                                         tx_entry->iov[index].iov_len,
548                                         FI_SEND, 0, 0, 0,
549                                         &tx_entry->mr[index], NULL);

```

Reply:
Function rxr_inline_mr_reg is re-formatted to fit for shm and efa cases.